crates/evm/core/src/utils.rs

```diff
@@ -5,48 +5,49 @@ use eyre::ContextCompat;
 use foundry_config::NamedChain;
 use revm::{
     interpreter::InstructionResult,
-    primitives::{Eval, Halt, SpecId, TransactTo},
+    primitives::{SpecId, TransactTo},
 };
 
 pub use foundry_compilers::utils::RuntimeOrHandle;
 pub use revm::primitives::State as StateChangeset;
+use revm::primitives::{HaltReason, SuccessReason};
 
 pub use crate::ic::*;
 
 /// Small helper function to convert an Eval into an InstructionResult
 #[inline]
-pub fn eval_to_instruction_result(eval: Eval) -> InstructionResult {
-    match eval {
-        Eval::Return => InstructionResult::Return,
-        Eval::Stop => InstructionResult::Stop,
-        Eval::SelfDestruct => InstructionResult::SelfDestruct,
+pub fn eval_to_instruction_result(reason: SuccessReason) -> InstructionResult {
```

Comment:
`SuccessReason` is `Into<InstructionResult>`: <https://docs.rs/revm/latest/revm/interpreter/primitives/enum.SuccessReason.html#impl-From%3CSuccessReason%3E-for-InstructionResult>